cleverpush/src/main/java/com/cleverpush/inbox/InboxDetailActivity.java

```diff
@@ -0,0 +1,574 @@
+    try {
+      View layout = activity.getLayoutInflater().inflate(layoutId, null);
+      layout.setOnClickListener(view -> dismiss());
+      return layout;
+    } catch (Exception exception) {
+      Logger.e(TAG, exception.getLocalizedMessage());
+    }
+    return null;
+  }
+
+  private void handleBundleData(Bundle extras) {
+    if (extras != null) {
+      if (extras.containsKey("selectedPosition")) {
+        selectedPosition = extras.getInt("selectedPosition");
+      }
+      if (extras.containsKey("notifications")) {
+        notifications = (ArrayList<Notification>) extras.getSerializable("notifications");
+        Notification clickedNotification = notifications.get(selectedPosition);
+        appBannerId = clickedNotification.getInboxAppBanner();
+        appBannerNotificationId = clickedNotification.getId();
+        showBannerById(appBannerId, appBannerNotificationId);
+      }
+    }
+  }
+
+  private void loadInboxDetails() {
+    body = findViewById(R.id.inboxViewBannerBody);
+    ImageView bannerBackGroundImage = findViewById(R.id.inboxViewBannerBackgroundImage);
+    ConstraintLayout parent = findViewById(R.id.parent);
+
+    runOnUiThread(new Runnable() {
+      @Override
+      public void run() {
+        composeBackground(parent, body);
+      }
+    });
+
+    isInitialized = true;
+
+    show();
+
+    setOpenedListener(action -> {
+//            sendBannerEvent("clicked", bannerPopup.getData());
+
+      if (getCleverPushInstance().getAppBannerOpenedListener() != null) {
+        getCleverPushInstance().getAppBannerOpenedListener().opened(action);
+      }
+
+      if (action.getType().equals("subscribe")) {
+        getCleverPushInstance().subscribe();
+      }
+
+      if (action.getType().equals("addTags")) {
+        getCleverPushInstance().addSubscriptionTags(action.getTags().toArray(new String[0]));
+      }
+
+      if (action.getType().equals("removeTags")) {
+        getCleverPushInstance().removeSubscriptionTags(action.getTags().toArray(new String[0]));
+      }
+
+      if (action.getType().equals("addTopics")) {
+        Set<String> topics = getCleverPushInstance().getSubscriptionTopics();
+        topics.addAll(action.getTopics());
+        getCleverPushInstance().setSubscriptionTopics(topics.toArray(new String[0]));
+      }
+
+      if (action.getType().equals("removeTopics")) {
+        Set<String> topics = getCleverPushInstance().getSubscriptionTopics();
+        topics.removeAll(action.getTopics());
+        getCleverPushInstance().setSubscriptionTopics(topics.toArray(new String[0]));
+      }
+
+      if (action.getType().equals("setAttribute")) {
+        getCleverPushInstance().setSubscriptionAttribute(action.getAttributeId(), action.getAttributeValue());
+      }
+
+      if (action.getType().equals("switchScreen")) {
+      }
+    });
+  }
+
+  public void show() {
+    if (!isInitialized) {
+      throw new IllegalStateException("Must be initialized");
```

Comment:
do we need to throw an exception here is error log and retun not enough? @gbhrdt ?